src/api/entities/Identity/index.ts

```diff
@@ -506,4 +510,75 @@ export class Identity extends Entity<UniqueIdentifiers> {
 
     return boolToBoolean(result);
   }
+
+  /**
+   * Retrieve every Dividend Distribution for which this Identity is eligible and hasn't been paid
+   *
+   * @note uses the middleware
+   * @note this query can be potentially **SLOW** depending on which Tokens this Identity has held
+   */
+  public async getPendingDistributions(): Promise<DistributionWithDetails[]> {
+    const { context, did } = this;
+    let tokens: SecurityToken[] = [];
+    let allFetched = false;
+    let start: number | undefined;
+
+    while (!allFetched) {
+      const { data, next } = await this.getHeldTokens({ size: MAX_PAGE_SIZE, start });
+      start = (next as number) || undefined;
+      allFetched = !next;
+      tokens = [...tokens, ...data];
+    }
+
+    const distributions = await this.context.getDividendDistributionsForTokens({ tokens });
+
+    const now = new Date();
+
+    /*
+     * We filter distributions out if:
+     *   - They have expired
+     *   - They have not begun
+     *   - They have not enough remaining funds to pay this Identity's share
+     *   - This Identity has already been paid
+     */
+    return P.filter(
+      distributions,
+      async ({ distribution, details: { remainingFunds } }): Promise<boolean> => {
+        const { expiryDate, perShare, ticker, id: localId, paymentDate } = distribution;
+
+        const isExpired = expiryDate && expiryDate < now;
+        const hasNotStarted = paymentDate > now;
+        const noFunds = remainingFunds.eq(0);
```

Comment:
We should remove this filter from here. Pending distributions should be there even if they have not enough remaining funds to pay. 
That's the only way the investor could know the problem.
We should add this check for paying or claiming dividends.